### DISCUSSION TITLE
Published another article explaining AvaloniaUI basic concepts

### DISCUSSION THREAD

**Discussion by npolyak** [AvaloniaUI/Avalonia#6541] [General]
The new article is available at [Multiplatform AvaloniaUI .NET Framework Programming Basic Concepts in Easy Samples](https://www.codeproject.com/Articles/5311995/Multiplatform-AvaloniaUI-NET-Framework-Programming). It provides detailed explanations and code samples for the most basic and important AvaloniaUI/WPF concepts, namely
1. Visual Trees
2. Logical Trees
3. Attached Properties
4. Style Properties
5. Direct Properties
6. Bindings
I'd greatly appreciate if someone can read and comment on this article telling me what can be improved and what other topics they want to be covered.

**Comment by grokys** [comment]
Excellent @npolyak ! Mind if we tweet this?

**Reply by npolyak** [reply]
Not at all - please go ahead

**Reply by grokys** [reply]
@npolyak messaged you on gitter.

**Comment by grokys** [comment]
The only suggestion so far is that the framework is actually called "Avalonia" - "AvaloniaUI" is the github org/company, but it's a nit and really not that important.

**Reply by npolyak** [reply]
I changed the latest article, I'll try to do the changes for the other articles on Sunday.

**Comment by santhoshkumarmk** [comment]
Can u explain same what you written for contextmenu from XAML perspective . How we can written from backend code to achieve same thing . I know in WPF so.

**Reply by npolyak** [reply]
Hey Santosh, 
context menu sample is given in the previous article [Multiplatform UI Coding with AvaloniaUI in Easy Samples. Part 1 - AvaloniaUI Building Blocks](https://www.codeproject.com/Articles/5308645/Multiplatform-UI-Coding-with-AvaloniaUI-in-Easy-Sa). 

Here is the XAML code:

    <Grid.ContextMenu>
        <ContextMenu>
           ...
        </ContextMenu>
    </Grid.ContextMenu>

this is equivalent to 

    Grid grid = ...;
    grid.ContextMenu = new ContextMenu();

**Reply by santhoshkumarmk** [reply]
How add Menuitems to this contextmenu ?

**Reply by npolyak** [reply]
List<MenuItem> menuItems = new List<MenuItem>();
    MenuItem saveMenuItem =  MenuItem{ Header = "Save" }; 
    // wire the command to the menu item 
    ...
    menuItems.Add(saveMenuItem);
    // add more menu items
    ...
    grid.ContextMenu.Items = menuItems;

**Reply by santhoshkumarmk** [reply]
This is my method

 private void connect(object sender, RoutedEventArgs e)
        {
            Grid grid = new Grid();
            grid.ContextMenu = new ContextMenu();
            List<MenuItem> menuItems = new List<MenuItem>();
            MenuItem saveMenuItem = new  MenuItem{ Header = "Save" };
menuItems.Add(saveMenuItem);
            grid.ContextMenu.Items = menuItems;
        }

I called like this from UI

<Button x:Name="btnalert" Foreground="Red" Content="Login" Click="connect" >
          
        </Button>

but contextmenu not appearing

**Reply by maxkatz6** [reply]
https://github.com/AvaloniaUI/Avalonia/discussions/6497

**Reply by npolyak** [reply]
I assume that the question moved to another thread.

**Reply by santhoshkumarmk** [reply]
Hi Nick,

In that there is no proper solution we already had discussion on that thread. can i expect any solution from your side ?

**Reply by npolyak** [reply]
Hey Santosh, in order to help you, I need to understand better what you want. The ContextMenu should not open on a button click - it usually opens on right mouse click

**Reply by maxkatz6** [reply]
In the link what I attached there is an example with Attached Flyout, which can be opened by any command (depending where you execute it). Visually it works the same as ContextFlyout, since that's its replacement, that can be used either as ContextFlyout or Attached Flyout.
https://github.com/AvaloniaUI/Avalonia/discussions/6497#discussioncomment-1257483
